src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -796,6 +798,9 @@ private static JsonRpcMessage CreateError(JToken id, Exception exception)
                     }
                 }
 
+                // Yield now so method invocation is async and we can proceed to handle other requests meanwhile
+                await Task.Yield();
```

Comment:
It looks like `await Task.Yield();` unconditionally captures the context of the caller. It appears you are introducing the first uses of it to this library, so you *could* be introducing deadlocks if any caller were to use `Task.Wait()` on the Task we return while on the UI thread. 
I can't find any immediate replacement for `Task.Yield()` that doesn't capture the `SynchronizationContext` (surprisingly, it appears there isn't one). We can add a private one and use it here, or we could [add one to the vs-threading library](https://github.com/Microsoft/vs-threading/issues/181).

Reply:
I know that Task.Yield() captures the context of the caller. In this case though it should be fine because the caller runs in a task on a background thread, started with ```Task.Run(this.ReadAndHandleRequestsAsync, this.disposeCts.Token);``` in [StartListening()](https://github.com/Microsoft/vs-streamjsonrpc/blob/master/src/StreamJsonRpc/JsonRpc.cs#L256)